### PR TITLE
[APPS-3029] Add manifest for rabbit-mq

### PR DESCRIPTION
**Pull request by kopylets**

### CODE REVIEW

File: RabbitMq/1.0.0/installer-manifest.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "add-array-option": [
+        {
+            "target": "\\Pyz\\Client\\RabbitMq\\RabbitMqConfig::getSynchronizationQueueConfiguration",
+            "source": "Spryker\\Shared\\AssetExternalStorage\\AssetExternalStorageConfig::ASSET_EXTERNAL_SYNC_STORAGE_QUEUE"
```
Comment (olehnovatskyi-spryker) [inline RabbitMq/1.0.0/installer-manifest.json:5]:
```suggestion
            "source": "\\Spryker\\Shared\\AssetExternalStorage\\AssetExternalStorageConfig::ASSET_EXTERNAL_SYNC_STORAGE_QUEUE"
```
I guess it should be fully qualified namespace to constant

Comment (olehnovatskyi-spryker) [inline RabbitMq/1.0.0/installer-manifest.json:5]: And we have a ticket where we should rename `AssetExternalStorageConfig` to `AssetStorageConfig` and other modules related to Assets: https://spryker.atlassian.net/browse/APPS-3066

File: RabbitMq/1.0.0/installer-manifest.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (olehnovatskyi-spryker) [inline RabbitMq/1.0.0/installer-manifest.json:1]: This file is the recipe for the AssetExternalStorage (AssetStorage) module.

**Review by olehnovatskyi-spryker — CHANGES_REQUESTED** [review]

**Comment by dereuromark** [comment]
Recipes should not be added manually but via release app
Whats the reason for this manual approach?

**Comment by olehnovatskyi-spryker** [comment]
I don't know how it does and have not heard anything about that.
@kopylets resolve this question. @dereuromark help with it, please. Because Rene is on vacation now and I guess only he can help with it from our stream.

File: RabbitMq/1.0.0/installer-manifest.json (hunk 2 above)
Comment (kopylets) [reply RabbitMq/1.0.0/installer-manifest.json:1]: Fixed

**Comment by kopylets** [comment]
> Recipes should not be added manually but via release app Whats the reason for this manual approach?

I guess I don't have an access to release processing. Should I ask someone with the right permissions to do this?

**Comment by dereuromark** [comment]
You dont have access to https://github.com/spryker/release-app/tree/master/src/Manifest ?
I can give you, just slack me.

The idea is that we try to always auto-create recipes with help of tooling, so we can roll this out for all our releases with clicks instead of high human error rate of manual process.

**Comment by kopylets** [comment]
> You dont have access to https://github.com/spryker/release-app/tree/master/src/Manifest ? I can give you, just slack me.
> 
> The idea is that we try to always auto-create recipes with help of tooling, so we can roll this out for all our releases with clicks instead of high human error rate of manual process.

Yes, I dont have access to https://github.com/spryker/release-app/tree/master/src/Manifest 

I'm afraid I can't directly slack you because I'm an external developer. It might work if you ping me. My account - Mikhail Kopylets, member id - U02NQH9778X

**Comment by dereuromark** [comment]
Please take care of https://github.com/spryker-sdk/integrator-recipes/pull/49 first
Finish PR and ping me. I will merge it along with release app changes then.

This has highest prio as the current structure is error prone.

File: RabbitMq/1.0.0/installer-manifest.json (hunk 1 above)
Comment (kopylets) [reply RabbitMq/1.0.0/installer-manifest.json]: I think we should do this as a separate refactoring ticket after renaming ticket is done. Also, I'm not sure if this manual addition of the manifest is still relevant after creating the appropriate strategy in the release app.

**Comment by kopylets** [comment]
It is currently not possible to apply this manifest because base class  - Spryker\Client\RabbitMq\RabbitMqConfig does not contain the getSynchronizationQueueConfiguration method. I think we should add this, and then all will be work fine.

**Review by oleksander-kiiashko — APPROVED** [review]

**Comment by stereomon** [comment]
> It is currently not possible to apply this manifest because base class - Spryker\Client\RabbitMq\RabbitMqConfig does not contain the getSynchronizationQueueConfiguration method. I think we should add this, and then all will be work fine.

How does it work in other cases? E.g. a Plugin should be added to a non-existent stack. IMO the stack will be created first by the manifest.

**Review by stereomon — APPROVED** [review]

**Comment by gechetspr** [comment]
Can we put on pause this one?

I don't like the idea that we have separate manifest key that actually doesn't reflect what it do. 

Also why can't we use existing `configure-module` to replace array completely?

**Comment by stereomon** [comment]
> Can we put on pause this one?

No, we need it.

> I don't like the idea that we have separate manifest key that actually doesn't reflect what it do.

Which `manifest key` you mean?

> Also why can't we use existing configure-module to replace array completely?

The array we want to add to may contain values we don't know, how should we replace it completely?

**Comment by gechetspr** [comment]
@stereomon manifest key `add-array-element` is very generic. It says nothing that we add array element for the config

> The array we want to add to may contain values we don't know, how should we replace it completely?
I mean we can generate manifest where we add a new return value for the config method. It will replace the old one

**Comment by dereuromark** [comment]
Maybe it should be called add-config-array-element

**Comment by gechetspr** [comment]
@dereuromark if we need this ASAP, I agree on this. I can change manifest generation in order to still use one strategy, but with different keys. I still need to make it work with edge cases and resolve conflicts. Currently it seems it can generate conflicting manifests and we don't resolve this.

@stereomon @kopylets I'll adjust manifest key in the RA. Please consider this change in the integrator and here

**Comment by gechetspr** [comment]
@stereomon `configure-module` supports arrays. It's possible that for now it doesn't compare values, but just methods, but we can fix this

**Comment by gechetspr** [comment]
Also a follow up should add a manifest type to remove array element

**Comment by kopylets** [comment]
> In any case let's do it with `add-config-array-element` manifest type.

Renamed.

**Comment by kopylets** [comment]
> Also a follow up should add a manifest type to remove array element

I think this is a big task that was not originally described. Can we make this a separate task?

**Comment by stereomon** [comment]
> I think this is a big task that was not originally described. Can we make this a separate task?

Yes, please create a follow-up for this.

**Review by gechetspr — APPROVED** [review]

**Pull request merged**